metaflow/plugins/deployer.py

```diff
@@ -0,0 +1,192 @@
+import os
+import sys
+import json
+import tempfile
+from typing import Optional, Dict, ClassVar
+
+from metaflow import Run, metadata
+from metaflow.exception import MetaflowNotFound
+from metaflow.runner.utils import clear_and_set_os_environ
+from metaflow.runner.subprocess_manager import CommandManager, SubprocessManager
+from metaflow.runner.utils import read_from_file_when_ready
+
+
+def handle_timeout(tfp_runner_attribute, command_obj: CommandManager):
+    try:
+        content = read_from_file_when_ready(tfp_runner_attribute.name, timeout=10)
+        return content
+    except TimeoutError as e:
+        stdout_log = open(command_obj.log_files["stdout"]).read()
+        stderr_log = open(command_obj.log_files["stderr"]).read()
+        command = " ".join(command_obj.command)
+        error_message = "Error executing: '%s':\n" % command
+        if stdout_log.strip():
+            error_message += "\nStdout:\n%s\n" % stdout_log
+        if stderr_log.strip():
+            error_message += "\nStderr:\n%s\n" % stderr_log
+        raise RuntimeError(error_message) from e
+
+
+def get_lower_level_group(api, top_level_kwargs: Dict, _type: str, name: Optional[str]):
+    if _type is None:
+        raise ValueError(
+            "Deployer doesn't have a 'type' to target. Please use a sub-class of Deployer."
+        )
+    if name is None:
+        return getattr(api(**top_level_kwargs), _type)()
+    return getattr(api(**top_level_kwargs), _type)(name=name)
+
+
+class TriggeredRun(object):
```

Comment:
Functionally, is there a difference between an `ExecutingRun` and a `TriggeredRun`?